### PR TITLE
Write parser for MCI using acornjs

### PR DESCRIPTION
**Pull request by thomastanck**

### CODE REVIEW

**Comment by thomastanck** [comment]
I need help finishing up this PR. I have no idea how to sanely throw errors from inside the library! I've scattered `throw new SyntaxError("blah")` around the place... Please advise on what I should do instead.

File: src/stdlib/__tests__/parser.ts [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+  const context = mockContext(4)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(JSON.stringify(context.errors, undefined, 2)).toBe('[]')
+    expect(obj).toMatchSnapshot()
+    expect(obj.status).toBe('finished')
+  })
+})
+
+test('Does not parse incomplete statements', () => {
```
Comment (openorclose) [inline src/stdlib/__tests__/parser.ts]: Should this be in the metacircular interpreter tests? It should already be checked in the `disallowed_syntax` tests. Same for the other syntax checks.

Comment (thomastanck) [reply src/stdlib/__tests__/parser.ts]: True, the negative tests are a bit redundant. I'll remove the negative tests (but keep the positive ones).

**Comment by openorclose** [comment]
I think your commits diallowing assignment expressions and update expressions should be in another pr.

**Comment by openorclose** [comment]
For the error throwing, I guess throwing SyntaxError is fine as they will still be caught? You can try providing the location of the error as well.

**Comment by thomastanck** [comment]
Split into #116 and #117

**Pull request closed without merging**